src/components/ec-table-filter/ec-table-filter.spec.js

```diff
@@ -87,6 +87,57 @@ describe('EcTableFilter', () => {
     expect(wrapper.element).toMatchSnapshot();
   });
 
+  it('should hide filters with "tw-hidden" class when "isHidden" is set', async () => {
+    const hiddenFilters = [
+      {
+        name: 'text',
+        component: markRaw(EcTextFilter),
+        isHidden: true,
+      },
+    ];
+    const wrapper = mountEcTableFilterAsTemplate(
+      '<ec-table-filter v-model="value" :filters="filters" />',
+      {},
+      {
+        data() {
+          return {
+            value: {},
+            filters: [...filters, ...hiddenFilters],
+          };
+        },
+      },
+    );
+    const dueDateFilter = wrapper.findByDataTest('ec-table-filter__filter-item-4');
+    expect(dueDateFilter.classes()).toContain('tw-hidden');
+    expect(dueDateFilter.element).toMatchSnapshot();
+
+    hiddenFilters[0].isHidden = false;
+    await wrapper.setProps({
+      filters: hiddenFilters,
+    });
+
+    expect(dueDateFilter.classes()).not.toContain('tw-hidden');
+    expect(dueDateFilter.element).toMatchSnapshot();
+  });
+
+  it('should hide the clear filters button when "isClearFiltersButtonHidden" is set', () => {
+    const wrapper = mountEcTableFilterAsTemplate(
+      '<ec-table-filter v-model="value" :filters="filters" />',
+      {
+        isClearFiltersButtonHidden: true,
+      },
+      {
+        data() {
+          return {
+            value: { feeType: modelValue.feeType },
+            filters,
+          };
+        },
+      },
+    );
+    expect(wrapper.element).toMatchSnapshot();
```

Comment:
let's narrow the snapshot or literally check whether clear filters button exists?

Reply:
I'm going to check if the button exists, should I do the same for the `ec-smart-table` test?

Reply:
Yeah, that snapshot is not too big, but we could do the same there